### PR TITLE
c++ API: Refactor statistics handling. 

### PR DESCRIPTION
**Pull request by aniemetz**
This moves term statistics collection from Solver to TermManager.
TermManager and Solver now both provide `getStatistics()`, the former
captures API level term statistics, the latter internal solver
statistics.

This also adds iterator capabilities to the Python Statistics class.

This builds on top of #10528.

### CODE REVIEW

File: include/cvc5/cvc5.h [hunk 1]
```diff
@@ -4161,6 +4189,10 @@ class CVC5_EXPORT TermManager
 
   /** The associated node manager. */
   internal::NodeManager* d_nm;
+  /** The statistics collected on the Api level. */
+  std::unique_ptr<APIStatistics> d_stats;
+  /** The statistics registry (indepent from any Solver's registry. */
```
Comment (ajreynol) [inline include/cvc5/cvc5.h:4194]:
```suggestion
  /** The statistics registry (independent from any Solver's registry). */
```

**Review by ajreynol — CHANGES_REQUESTED** [review]
Can you rebase? 
Also, can you comment on whether there are any expected behavior changes when using CLI?

**Comment by aniemetz** [comment]
> Also, can you comment on whether there are any expected behavior changes when using CLI?

As mentioned in the description, term related statistics are now captured in the TermManager statistics and not included in the Solver statistics anymore (everything we capture with `increment_*_stats).

**Comment by ajreynol** [comment]
So we don't print TermManager statistics via CLI using `--stat-internal`?  If not, should we modify `src/main/` to do this?

**Comment by aniemetz** [comment]
Good point, I guess so?

**Comment by aniemetz** [comment]
Wait, shouldn't this already be taken care of via `CommandExecutor::printStatisticsSafe()`?

**Comment by aniemetz** [comment]
Addressed.

**Review by ajreynol — APPROVED** [review]

**Pull request merged**